docs/user-guide/petset.md

```diff
@@ -0,0 +1,370 @@
+
+```
+                    [ init container ]                   [ app container ]
+    list of peers -> | on-start.sh  | -> valid db.conf -> reload database
+```
+
+Most admins will be able to produce a valid `on-start.sh` and/or `on-change.sh`. To help with the listing of peers, we can run the [peer finder](https://github.com/kubernetes/contrib/tree/master/pets/peer-finder) in an init container. Since init containers share the hostname of the Pet, the peer finder can wait till it sees itself in DNS, then invoke your `on-start.sh` with a list of peers. Your `on-start.sh` is a black box that produces a valid db.conf, it might have to actually ping peers for this, it might even have to join a cluster to check the transaction id. The only invariants are that before the app container starts it must have a:
+
+1. up to date data directory
+2. valid config file
+
+The following examples implement various forms of this pattern for different database deployments.
+
+* [redis](https://github.com/kubernetes/kubernetes/tree/master/test/e2e/testing-manifests/petset/redis) master/slave cluster
+* [zookeeper](https://github.com/kubernetes/kubernetes/tree/master/test/e2e/testing-manifests/petset/zookeeper) quorum based cluster
+* [mysql galera](https://github.com/kubernetes/kubernetes/tree/master/test/e2e/testing-manifests/petset/mysql-galera) active-active cluster
+
+## Troubelshooting
+
+You might have noticed an `annotations` field in all the Pet Sets shown above.
+
+```yaml
+annotations:
+  pod.alpha.kubernetes.io/initialized: "true"
+```
+
+This field is a debugging hook. It pauses any scale up/down operations on the entire Pet Set. If you'd like to pause a petset after each pet, set it to `false` in the template, wait for each pet to come up, verify it has initialized correctly, and then set it to `true` using `kubectl edit` on the pet (setting it to `false` on *any pet* is enough to pause the Pet Set). If you don't need it, create the Pet Set with it set to `true` as shown. This is surprisingly useful in debugging bootstrapping race conditions.
+
+## Future Work
+
+There are a LOT of planned improvements since Pet Set is still in alpha.
+
+* Data gravity and local storage
+* Richer notification events
+* Public network identities
+* WAN cluster deployments (multi-AZ/region/cloud provider)
+* Image and node upgrades
+
+This list goes on, if you have examples, ideas or thoughts, please contribute.
+
+## Alternatives
+
+Deploying one RC/Service per pod is a popular alternative, as is simply deploying a DaemonSet that utilizes the identity of a Node. Entrypoint scripts are commonly used instead of init containers.
```

Comment:
You may want to note that an RC with scale=1 does not mean there will only ever be one pod (we start creation immediately when the old one is requested to be deleted, not when it's actually deleted).